File: src/main/java/org/hoohoot/homelab/manager/notifications/matrix/MatrixMessage.java
package org.hoohoot.homelab.manager.notifications.matrix;

import com.fasterxml.jackson.annotation.JsonProperty;

public record MatrixMessage(
        @JsonProperty("msgtype") String msgType,
        String body,
        String format,
        @JsonProperty("formatted_body") String formattedBody
) {
    public static MatrixMessage html(String body) {
        return new MatrixMessage("m.text", body, "org.matrix.custom.html", body);
    }
}

File: src/main/kotlin/org/hoohoot/homelab/manager/notifications/matrix/MatrixMessage.kt
package org.hoohoot.homelab.manager.notifications.matrix

import com.fasterxml.jackson.annotation.JsonProperty

data class MatrixMessage(
    @field:JsonProperty("msgtype") @param:JsonProperty("msgtype") val msgType: String,
    val body: String,
    val format: String,
    @field:JsonProperty("formatted_body") @param:JsonProperty("formatted_body") val formattedBody: String
) {
    companion object {
        @JvmStatic
        fun html(body: String): MatrixMessage {
            return MatrixMessage("m.text", body, "org.matrix.custom.html", body)
        }
    }
}